### PR TITLE
fix: estimate gas

### PR DESCRIPTION
**Pull request by antazoey**
### What I did

NOTE: Is only a POV - I havent tested yet.
fixes: #36 

### Checklist

- [ ] Passes all linting checks (pre-commit and CI jobs)
- [ ] New test cases have been added and are passing
- [ ] Documentation has been updated
- [ ] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

**Comment by fubuloubu** [comment]
This seems to work well for me, but needs some update to the mock provider

**Comment by solarthesis** [comment]
does this fix the need to provide `gas=0` ?

File: ape_safe/client/mock.py [hunk 1]
```diff
@@ -111,5 +111,5 @@ def post_signatures(
             )
 
     def estimate_gas_cost(self, receiver: AddressType, value: int, data: bytes, operation: int = 0):
-        # TODO
-        return 0
+        # Estimate normally.
+        return NotImplemented
```
Comment (fubuloubu) [inline ape_safe/client/mock.py:115]: Wait, I don't think returning an error is the right way to go here

Comment (antazoey) [reply ape_safe/client/mock.py:115]: It is a constant, not an error, but feel free to change.
https://stackoverflow.com/questions/1062096/python-notimplemented-constant

Comment (antazoey) [reply ape_safe/client/mock.py:115]: alright alright we'll just have it make the request normally or something.
The weird part is that Safe's API is like different a bit than just normal eth rpc stuff

**Comment by antazoey** [comment]
the tests fail on main for the same reason; not sure what changed or why that suddenly happens.

**Comment by antazoey** [comment]
shipppppp itttttttttttt

**Comment by solarthesis** [comment]
this is never gonna ship @antazoey forget about it

**Comment by antazoey** [comment]
> this is never gonna ship @antazoey forget about it

why do you say this?

**Comment by antazoey** [comment]
https://giphy.com/clips/anidom-the-simpsons-season-33-ep-13-s33e13-aGkBFXQnPN6McpWtkE

File: ape_safe/client/mock.py [hunk 2]
```diff
@@ -109,3 +109,8 @@ def post_signatures(
                     signatureType=SignatureType.EOA,
                 )
             )
+
+    def estimate_gas_cost(
+        self, receiver: AddressType, value: int, data: bytes, operation: int = 0
+    ) -> Optional[int]:
+        return None  # Estimate gas normally
```
Comment (fubuloubu) [inline ape_safe/client/mock.py:116]: I'm assuming this works in the tests, but if not I think the issue is likely that the gas prices are nonzero

**Review by fubuloubu — APPROVED** [review]

File: ape_safe/client/mock.py (hunk 2 above)
Comment (antazoey) [reply ape_safe/client/mock.py:116]: What issue?

**Pull request merged**